### PR TITLE
Guard against invalid-mag bucket volume save actions

### PR DESCRIPTION
**Pull request by fm3**
This PR ensures that the front-end doesn't try to save buckets for magnifications which don't exist. This caused save errors before (without any helpful message).
The scenario only happened when layers with differing mags existed (e.g., mag 2-2-2 vs mag 2-2-1) which is not supported, anyway. However, the save logic used layer-independent resolutions which is not the correct way of handling this, which is why I took the opportunity to clean this up a bit.

### URL of deployed dev instance (used for testing):
- https://volumemagerror.webknossos.xyz

### Steps to test:
- The actual error scenario is hard to reproduce (the same dataset worked for @fm3 but not for me). I'd suggest to test the happy paths instead, since these are far more important.
- Create an annotation and brush (also use floodfill, since a bit of code changed there) --> save and reload
- Create another annotation (e.g., with layer restrictions) and test again

### Issues:
- https://scm.slack.com/archives/C5AKLAV0B/p1667468655289759
- related to https://github.com/scalableminds/webknossos/issues/6604

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ class VolumeTracingService @Inject()(
               case a: UpdateBucketVolumeAction =>
                 if (tracing.getMappingIsEditable) {
                   Fox.failure("Cannot mutate volume data in annotation with editable mapping.")
-                } else updateBucket(tracingId, tracing, a, updateGroup.version)
+                } else updateBucket(tracingId, tracing, a, updateGroup.version) ?~> "Failed to save volume data."
```
Comment (philippotto) [inline webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala:98]: When I provoked the error, I only saw this message here and not the "received a mag...". is this intentional? should there be an error chain?

File: frontend/javascripts/test/model/binary/layers/wkstore_adapter.spec.ts [hunk 2]
```diff
@@ -15,12 +15,18 @@ const RequestMock = {
 const { dataSource } = datasetServerObject;
 let _fourBit = false;
 
+// test
+
```
Comment (philippotto) [inline frontend/javascripts/test/model/binary/layers/wkstore_adapter.spec.ts:19]:
```suggestion
```

**Review by daniel-wer — APPROVED** [review]
Code LGTM :+1: 

I encountered one issue during testing:
- [x] When restricting the volume annotation (for example, to start at 4-4-4) and then using the quick select tool in 1-1-1, the following error toast is shown: "Error: Magnification with index 0 does not exist.". This might be alright, but I wanted to mention it since all other annotation tools work in this mag.

**Comment by philippotto** [comment]
> [x] When restricting the volume annotation (for example, to start at 4-4-4) and then using the quick select tool in 1-1-1, the following error toast is shown: "Error: Magnification with index 0 does not exist.". This might be alright, but I wanted to mention it since all other annotation tools work in this mag.

Thanks for catching this! The issue was unrelated to the PR, but I fixed it, anyway :) See my last commit.

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala (hunk 1 above)
Comment (fm3) [reply webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala:98]: I think there should be a chain. I’ll have another look

Comment (fm3) [reply webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala:98]: You were right, there was a conversion that ate the error chain. Should be fixed now :)

**Comment by fm3** [comment]
I’d say the backend changes are simple enough that they don’t need another review. @philippotto assuming you are happy with the front-end, feel free to merge :)

**Pull request merged**